### ISSUE TITLE
Bind to multiple sockets?

### ISSUE THREAD

**Issue by timbru** [hyperium/hyper#3009]
First off.. apologies for the noise if this is already possible/discussed/dismissed. But, I could not find out how to do this.

My use case is that I would like the hyper server to bind to both IPv4 and IPv6. I can imagine there might be similar use cases where a host has multiple public IP addresses and one wants a hyper server to be available on more than one.

**Is your feature request related to a problem? Please describe.**
Users are reporting issues with happy eyeball connections to a host, if the host has IPv6, but hyper only listens on IPv4.

**Describe the solution you'd like**
I'd like to bind to both IPv4 and IPv6.

**Describe alternatives you've considered**
Our current advice to users of our software is that they should just let it listen to 127.0.0.1 (IPv4 localhost, which is the default we use) and use a proxy server in front of it if they want public access. Proxy servers also offer great support for HTTPS, access restrictions, etc etc.

But, while we actually believe that using a proxy server (and IPv4 localhost for our software) is best practice some users do not want to use a proxy.

**Comment by seanmonstar** [comment]
It should already be possible, but perhaps hard to find a good example. In general, you can `TcpStream::bind` with the IPv4, and spawn a task using that with `Server` or (`server::conn::Http`), and then do the same with the IPv6 address. You'd thus have 2 "accepting" tasks.

**Comment by timbru** [comment]
Yeah, that makes sense. Having an example would be useful, but I think I have enough to give this a try.
